### PR TITLE
IMPRO-1815: reject float64 data in combine plugin

### PR DESCRIPTION
**Pull request by MoseleyS**
The CubeCombiner plugin now throws a `TypeError` if the specified combine operation results in float64 data.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Review by BelligerG — DISMISSED** [review]
Good to go :-)

File: improver/cube_combiner.py [hunk 1]
```diff
@@ -277,6 +277,14 @@ def process(
...
             result.data = result.data / len(cube_list)
 
+        # Check resulting dtype and modify if necessary
+        if result.dtype == np.float64:
+            unique_cube_types = set([c.dtype for c in cube_list])
+            raise TypeError(
+                f"Operation {self.operation} on types {unique_cube_types} results in "
+                "float64 data which cannot be safely coerced to float32"
+            )
+
```
Comment (cgsandford) [inline improver/cube_combiner.py:287]: Could you add some information to the docstring please: at least to the `Raises` section.  Users expecting this plugin to be generic are going to be confused by this error.

**Review by cgsandford — COMMENTED** [review]
All tests pass, I'm happy except for the docstring contents.

File: improver/cube_combiner.py (hunk 1 above)
Comment (MoseleyS) [reply improver/cube_combiner.py:288]: Thanks. I'd missed this. Added to doc-string.

**Review by cgsandford — APPROVED** [review]

**Pull request merged**